rfcs/react-components/convergence/triggers-and-multiple-layers-of-button-handling.md

```diff
@@ -0,0 +1,124 @@
+# RFC: Triggers and multiple layers of button handling
+
+---
+
+_@bsunderhus_
+
+## Summary
+
+The trigger API is an elegant solution for the problem of how to provide properties to a button-like element.
+
+Currently most triggers will ensure its direct children has the proper handler and properties of a button, by using `useARIAButtonProps` behind the scenes.
+
+In the case of a `div` as triggers child, `useARIAButtonProps` will ensure that this `div` will have proper `button` ARIA attributes
+
+```tsx
+<Trigger>
+  <div>Trigger something</div>
+</Trigger>
+```
+
+```html
+<div onclick="fn" onkeydown="fn" onkeyup="fn" role="button" tabindex="0">Trigger something</div>
+```
+
+`useARIAButtonProps` also ensure that on the case a `button` as trigger children, then no extra attributes are necessary.
+
+```tsx
+<Trigger>
+  <button>Trigger something</button>
+</Trigger>
+```
+
+```html
+<button>Trigger something</button>
+```
+
+## Problem statement
+
+On the case of a custom component on the other hand, it's impossible for `useARIAButtonProps` to find out if the returned element will be an actual button or not.
+
+```tsx
+<Trigger>
+  <Button>Trigger something</Button>
+</Trigger>
+```
+
+That way, even for `Button`, `useARIAButtonProps` will end up providing a bunch of unnecessary props.
```

Comment:
Is it the problem? Will be something broken?

Reply:
Yup, that's the problem, nothing will be broken. It's just a bunch of redundant listeners and properties such as `role` and `tabIndex`

Reply:
But redundant properties will break nothing, correct?

Reply:
Yes, nothing will be broken, they'll just override one another.

Reply:
Should be then a third option in RFC called "Do nothing"? 🤔

Reply:
we will need to make sure that when tabindex is used explicitly, it won't be overriden by trigger:
<Trigger><Button role="menuitem" tabIndex={-1} /></Trigger> should still render a <button> with tabindex -1 and menuitem role.

Reply:
similarly keydown, keyup and click handlers on button should be respected